src/utils/maak_protocol.R

```diff
@@ -0,0 +1,263 @@
+#' @title Create a folder with a markdown template to start working on a new fieldwork protocol
+#'
+#' @description This function will create a new folder based on values that are passed on via the parameters and creates a R-markdown skeleton based on a template file to start working on a new protocol.
+#'
+#' @param titel
+#' @param subtitel
+#' @param auteurs
+#' @param datum
+#' @param reviewers
+#' @param documentbeheerder
+#' @param revisie
+#' @param projectnaam
+#' @param thematiek
+#'
+#' @return A new subfolder beneath src with an Rmarkdown file
+#'
+#' @importFrom rprojroot find_root
+#' @importFrom rprojroot is_git_root
+#' @importFrom whisker whisker.render
+#' @importFrom purrr map_chr
+#' @importFrom stringr str_replace_all
+#' @importFrom assertthat
+#' assert_that
+#' is.string
+#' is.date
+#' @examples
+#' maak_protocol()
+#'
+maak_svp <- function(
+  titel = "Titel",
+  subtitel = "",
+  korte_titel = "korte titel",
+  auteurs = "voornaam1 naam1, voornaam2 naam2, ...",
+  datum = Sys.Date(),
+  reviewers = "voornaam1 naam1, voornaam2 naam2, ...",
+  documentbeheerder = "voornaam naam",
+  revisie = "0.1",
+  projectnaam,
+  thematiek = c("generiek", "water", "lucht", "bodem", "vegetatie", "soorten")) {
+  
+  # needed for as long as this function is not part of a package
+  require(assertthat)
+  require(rprojroot)
+  require(whisker)
+  require(purrr)
+  require(stringr)
+ 
+  
+  # check parameters
+  assert_that(is.string(titel))
+  assert_that(is.string(subtitel))
+  assert_that(is.string(korte_titel) & length(korte_titel) <= 20)
+  assert_that(is.date(as.Date(datum)))
+  assert_that(is.string(auteurs))
+  assert_that(is.string(reviewers))
+  assert_that(is.string(documentbeheerder))
+  assert_that(is.string(revisie))
+  if (!missing(projectnaam)) {
+    assert_that(is.string(projectnaam))
+  }
+  thematiek <- match.arg(thematiek)
+  assert_that(is.string(thematiek))
+  
+  # template folder
+  project_root <- find_root(is_git_root)
+  path_to_template <- file.path(project_root, "src", "utils", "template")
+  
+  # create protocol name
+  
+  protocol_type <- "svp"
```

Comment:
I did that initially, but abandoned that idea because I think the different protocol types (svp, sip, sap, sop) will have different templates in the end (for now, I simply copy-pasted the one for svp). It is easier to write separate functions in that case and perhaps consider an overarching function later on.